dns_server/src/main.rs

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) 2023 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::Arc;
+
+use clap::Parser;
+use config::DnsServerConfig;
+use crawler::{storage_impl::DnsServerStorageImpl, Crawler};
+use p2p::net::default_backend::{
+    transport::{NoiseEncryptionAdapter, NoiseTcpTransport, TcpTransportSocket},
+    DefaultNetworkingService,
+};
+use tokio::sync::mpsc;
+
+mod config;
+mod crawler;
+mod error;
+mod server;
+
+async fn run(config: Arc<DnsServerConfig>) -> Result<void::Void, error::DnsServerError> {
+    let (command_tx, command_rx) = mpsc::unbounded_channel();
+
+    let stream_adapter = NoiseEncryptionAdapter::gen_new();
```

Comment:
Please do all the instantiations of p2p in a separate function and take the result and use it. There seems to be many steps that shouldn't be done in here.

Reply:
OK, updated.